primitives/src/lib.rs

```diff
@@ -1,19 +1,149 @@
-pub type CollectionId = u32;
-// Id used for identifying non-fungible items.
-pub type ItemId = u32;
-/// The maximum length of an attribute key.
-pub type KeyLimit = ConstU32<64>;
-/// The maximum approvals an item could have.
-pub type ApprovalsLimit = ConstU32<20>;
+#[derive(Encode, Decode, Debug, MaxEncodedLen, Eq, PartialEq)]
+#[cfg_attr(feature = "std", derive(TypeInfo))]
+pub struct AccountId(pub [u8; 32]);
+
+// Identifier for the class of asset.
+pub type AssetId = u32;
+
+#[cfg(feature = "nfts")]
+pub mod nfts {
+	use bounded_collections::ConstU32;
+
+	// Id used for identifying non-fungible collections.
+	pub type CollectionId = u32;
+	// Id used for identifying non-fungible items.
+	pub type ItemId = u32;
+	/// The maximum length of an attribute key.
+	pub type KeyLimit = ConstU32<64>;
+	/// The maximum approvals an item could have.
+	pub type ApprovalsLimit = ConstU32<20>;
+}
+
+pub mod v0 {
+	use super::*;
+	pub mod error {
+		use super::*;
+
+		#[derive(Encode, Decode, Debug, Eq, PartialEq)]
+		#[cfg_attr(feature = "std", derive(TypeInfo))]
+		#[repr(u8)]
+		pub enum Error {
+			/// Some unknown error occurred. Go to the Pop API docs section `Pop API error`.
+			Other {
+				// Index within the `DispatchError`
+				dispatch_error_index: u8,
+				// Index within the `DispatchError` variant.
+				error_index: u8,
+				// Index for further nesting, e.g. pallet error.
+				error: u8,
+			} = 0,
+			/// Failed to lookup some data.
+			CannotLookup = 1,
+			/// A bad origin.
+			BadOrigin = 2,
+			/// A custom error in a module.
+			Module { index: u8, error: u8 } = 3,
+			/// At least one consumer is remaining so the account cannot be destroyed.
+			ConsumerRemaining = 4,
+			/// There are no providers so the account cannot be created.
+			NoProviders = 5,
+			/// There are too many consumers so the account cannot be created.
+			TooManyConsumers = 6,
+			/// An error to do with tokens.
+			Token(TokenError) = 7,
+			/// An arithmetic error.
+			Arithmetic(ArithmeticError) = 8,
+			/// The number of transactional layers has been reached, or we are not in a transactional
+			/// layer.
+			Transactional(TransactionalError) = 9,
+			/// Resources exhausted, e.g. attempt to read/write data which is too large to manipulate.
+			Exhausted = 10,
+			/// The state is corrupt; this is generally not going to fix itself.
+			Corruption = 11,
+			/// Some resource (e.g. a preimage) is unavailable right now. This might fix itself later.
+			Unavailable = 12,
+			/// Root origin is not allowed.
+			RootNotAllowed = 13,
+			/// Unknown function id.
+			UnknownFunctionId = 254,
+			/// Decoding failed on the runtime.
+			DecodingFailed = 255,
+		}
+
+		impl From<u32> for Error {
+			fn from(value: u32) -> Self {
+				let encoded = value.to_le_bytes();
+				Error::decode(&mut &encoded[..]).unwrap_or(Error::DecodingFailed)
+			}
+		}
+
+		impl From<Error> for u32 {
+			fn from(value: Error) -> Self {
+				let mut encoded_error = value.encode();
```

Comment:
This is used for 
```
impl From<Error> for StatusCode {
	fn from(value: Error) -> Self {
		StatusCode::from(u32::from(value))
	}
}
```
in `pop-api/src/lib.rs`. Currently I'm only using it in tests, but I thought it might be useful for devs. Now that I think about it again, devs should never use this in production contracts. They should use this during development / debugging, not sure whether that is reason enough to feature gate it with `tests`?

Reply:
After revision this needs to be fixed. Thanks for pointing out.

Reply:
see commit "fix: replace test code to api fungibles"